### ISSUE TITLE
How to retrieve cookies in response message of a WebView

### ISSUE THREAD

**Issue by WenningQiu** [dotnet/maui#16266]
This might be a silly question but I am new to MAUI and need some help.

My application needs to retrieve the cookies in the response message of a WebView. I tried WebView.Cookies but it is empty when Navigated event is dispatched. So what is the right way to retrieve cookies? 

```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="WarehouseManager.UI.Views.AuthenticationView"
             xmlns:viewmodels="clr-namespace:WarehouseManager.UI.ViewModels"
             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
             BackgroundColor="White"
             Shell.BackgroundColor="DarkRed"
             Shell.NavBarIsVisible="False"
             Shell.FlyoutBehavior="Disabled"
             Title="AuthenticationView">
    <Shell.BackButtonBehavior>
        <BackButtonBehavior IsVisible="False" IsEnabled="False" />
    </Shell.BackButtonBehavior>
    <WebView x:Name="AuthWebView" Source="https://myserver/SsoService/login" Navigated="AuthWebView_OnNavigated" />
</ContentPage>
```
```
public partial class AuthenticationView : ContentPage
{
    private INavigationService _navigationService;

    public AuthenticationView(INavigationService navigationService)
    {
        InitializeComponent();
        _navigationService = navigationService;

        var webView = Content as WebView;

        if (webView.Cookies is null)
            webView.Cookies = new CookieContainer();
    }

   private async void AuthWebView_OnNavigated(object sender, WebNavigatedEventArgs e)
    {
        if (e.Result == WebNavigationResult.Success)
        {
            var webView = sender as WebView;

            foreach (var x in webView.Cookies.GetAllCookies())
            {
                Console.WriteLine(x);
            }
     }
}

```

**Comment by WenningQiu** [comment]
@mhrastegari  I upgraded CommunityToolkit.Maui to 5.2.0, but that does not help.

**Comment by mhrastegari** [comment]
This does nothing to do with MCT! 
You need to update your .NET and get the latest .NET MAUI Workloads.

**Comment by WenningQiu** [comment]
I updated my VS2022 to 17.6.5 which should give me the latest .NET MAUI workloads. But still got the same result.

**Comment by mhrastegari** [comment]
To be honest I'm not sure or didn't know that VS updates them automatically! Because it always asks me to download them.
And I really don't recommend to always rely on VS! So check it your self and update your .NET if it's not the latest version.

**Comment by WenningQiu** [comment]
VS appears to have got the latest (7.0.9) installed. @mhrastegari, do you by any chance remember the issues or sample code about accessing cookies in response messages? Perhaps I am just not using the right API.

```
.NET SDK:
 Version:   7.0.306
 Commit:    f500069cb7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.306\

Host:
  Version:      7.0.9
  Architecture: x64
  Commit:       8e9a17b221

.NET SDKs installed:
  7.0.306 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by Woselko** [comment]
I think this is similar issue to my: https://github.com/dotnet/maui/issues/16409

**Comment by jfversluis** [comment]
I think this should be fixed in the most recent versions, see #15405

If that is not the case, please open a new issue with any new insights you might have by now.

Thanks!
